msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java

```diff
@@ -3112,4 +3112,17 @@ private boolean isUseAnonymousId(Topic topic)
 		// Condenses to:
 		return topic.getPostAnonymous() && (!topic.getRevealIDsToRoles() || !uiPermissionsManager.isIdentifyAnonAuthors((DiscussionTopic) topic));
 	}
+
+    public int getPercentRead()
+    {
+        int totalPosts = 0;
+        List<Object[]> studentAuthoredStats = messageManager.findAuthoredMessageCountForAllStudents();
+        for (Object[] pair : studentAuthoredStats)
+        {
+            totalPosts += ((Long) pair[1]).intValue();
+        }
```

Comment:
This can be reduced to 1 line:

`totalPosts = studentAuthoredStats.stream().map( (pair) -> ((Long) pair[1]).intValue() ).reduce( totalPosts, Integer::sum );`

However, I'm not sure if it's more or less readable.